data-conversion/src/main.rs

```diff
@@ -1,104 +1,174 @@
-        "hex".to_string(),
-        "fixed".to_string(),
-    ];
-    // Array of all possible options to convert to
-    let to_types: Vec<String> =
-        vec!["binary".to_string(), "float".to_string(), "hex".to_string()];
-
-    if !from_types.contains(&args.ftype) {
-        panic!("{} is not a valid type to convert from", args.from);
+#[derive(Debug, PartialEq)] // Add PartialEq derivation here - What is this?
+enum NumType {
+    Binary,
+    Float,
+    Hex,
+    Fixed,
+}
+
+impl ToString for NumType {
+    fn to_string(&self) -> String {
+        match self {
+            NumType::Binary => "binary".to_string(),
+            NumType::Float => "float".to_string(),
+            NumType::Hex => "hex".to_string(),
+            NumType::Fixed => "fixed".to_string(),
+        }
     }
-    if !to_types.contains(&args.totype) {
-        panic!("{} is not a valid type to convert to", args.to);
+}
+
+impl FromStr for NumType {
+    type Err = ParseNumTypeError;
+
+    fn from_str(input: &str) -> Result<NumType, Self::Err> {
+        match input {
+            "binary" => Ok(NumType::Binary),
+            "float" => Ok(NumType::Float),
+            "hex" => Ok(NumType::Hex),
+            "fixed" => Ok(NumType::Fixed),
+            _ => Err(ParseNumTypeError),
+        }
     }
+}
+
+#[derive(FromArgs)]
+/// get arguments to convert
+struct Arguments {
+    /// file to convert from
+    #[argh(option)]
+    from: String,
+
+    /// file to convery to
+    #[argh(option)]
+    to: String,
+
+    /// type to convert from
+    #[argh(option)]
+    ftype: NumType,
+
+    /// type to convert to
+    #[argh(option)]
+    totype: NumType,
+
+    /// optional exponent for fixed_to_binary -> default is -1
+    #[argh(option, default = "-1.")]
+    exp: f32,
+}
+
+fn main() {
+    let args: Arguments = argh::from_env();
 
-    convert(&args.from, &args.to, &args.ftype, &args.totype);
+    convert(&args.from, &args.to, &args.ftype, &args.totype, args.exp);
 }
 
 /// Converts [filepath_get] from type [convert_from] to type
 /// [convert_to] in [filepath_send]
+///
+/// # Arguments
+///
+/// * `float_string` - A string slice containing the floating-point number to be converted.
+/// * `filepath_send` - A mutable reference to a `File` where the binary representation
+///   will be appended.
+///
+/// # Returns
+///
+/// Returns `Ok` if the operation is successful, or an `Err` if an I/O error occurs while
+/// writing to the file.
+///
+/// # Panics
+///
+/// Panics if the input string cannot be parsed as a floating-point number.
+///
+/// # Example
+///
+/// ```rust
+/// use std::fs::File;
+///
+/// let mut file = File::create("output.txt").expect("Failed to create file");
+/// float_to_binary("3.14", &mut file).expect("Failed to convert float to binary");
+/// ```
 fn convert(
     filepath_get: &String,
     filepath_send: &String,
-    convert_from: &String,
-    convert_to: &String,
+    convert_from: &NumType,
+    convert_to: &NumType,
```

Comment:
This is a pretty minor thing but could clean things up a tiny bit: if you put `#[derive(Clone, Copy)]` on `NumType`, you can pass it around without using `&` references and `*` dereferencing everywhere and everything will just work.

Longer explanation: the `Copy` trait gives the Rust compiler permission to silently copy stuff from place to place when passing it into functions, etc., instead of moving it (or requiring references/borrowing). This is not the default because it's a really bad idea for types that can get large. Imagine the performance frustration if Rust let you implicitly copy around objects that are, like, 1 MB in size! (Actually, no need to imagine: you can just try programming in C++, which will happily do this for you silently.)

Anyway, a simple `enum` like `NumType` is very small. In fact, it's definitely going to be no larger than a pointer to a `NumType`! So it's probably at least as cheap (and less annoying, requiring fewer `&`s and `*`s) to copy those values around instead of passing around references.